src/HotChocolate/Core/src/Execution/Processing/MiddlewareContext.Pure.cs

```diff
@@ -87,6 +87,8 @@ public T Parent<T>()
             {
                 T casted => casted,
                 null => default!,
+                object[] o when Temp.ValueConverter.TryGetValue(_selection.Id, out var converter) =>
```

Comment:
> So i added a few comments and explained a few things.
> 
> generally:
> 
> * Its important to know that this feature will only be used for one provider of data. Meaning, this will probably only work with ef core. I never tested it with other IQueryable providers but doubt that they would work in the same way. EfCore is really good in IQueryable translations
> * All changes to the core and execution are high impact changes. meaning the less changes in the core you do in a PR, the less work is involved in review and regression testing, the more likely it is the PR is merged
> * This feature is for IQueryable (probably only ef core) of the proejction engine. Most of the users do not use projections and not all of the projections user are using EF Core. Meaning, no change in this PR should ever surface as public API change on one of the Core APIs or break an existing projection API.
> 
> I guess we should have a chat together ( also with @michaelstaib when he is back from oslo) to define how we want to proceed here and make a plan. We have to have a common idea how we want this feature to be otherwise the risk is very high that this change will never land in main.

Right, so I guess I'm going to revert the commits with Core changes then.